pandas/io/packers.py

```diff
@@ -214,6 +217,14 @@ def convert(values):
     if dtype == np.object_:
         return v.tolist()
 
+    if is_categorical_dtype(dtype):
+        return {
```

Comment:
this can effectively be moved to ``encode`` with a type of ``Categorical``, rather than here